### PR TITLE
build: update vendored six from 1.11.0 to 1.16.0

### LINKED ISSUE #2282: ImportWarning on Python 3.10
When using Python 3.10, kafka-python 2.0.2, in `python -Wd`:

```
>>> from kafka import KafkaConsumer
<frozen importlib._bootstrap>:914: ImportWarning: _SixMetaPathImporter.find_spec() not found; falling back to find_module()
<frozen importlib._bootstrap>:671: ImportWarning: _SixMetaPathImporter.exec_module() not found; falling back to load_module()
```

Needs find_spec and exec_module here https://github.com/dpkp/kafka-python/blob/f19e4238fb47ae2619f18731f0e0e9a3762cfa11/kafka/vendor/six.py#L170

### PR DESCRIPTION
**Pull request by shifqu**
In this commit, the del X is still commented out due to the fact that upstream https://github.com/benjaminp/six/pull/176 is not merged.

closes #2282

- - -
This change is [Reviewable](https://reviewable.io/reviews/dpkp/kafka-python/2398)

### CODE REVIEW

File: kafka/vendor/six.py
Comment (shifqu) [inline kafka/vendor/six.py:664]: A new `del` statement. Does the same condition here apply as in https://github.com/dpkp/kafka-python/pull/979#discussion_r100403389?

**Comment by gpapia** [comment]
kafka-python 2.0.2 doesn't work anymore with Python 3.12 (why?) without this PR. It should be great to merge it.

**Comment by pawel-swiecki-saucelabs** [comment]
@gpapia Yes, this MR will most likely resolve ModuleNotFoundError, see https://github.com/dpkp/kafka-python/issues/2401.

**Comment by wbarnha** [comment]
Thank you for the PR, sorry I didn't notice this sooner. I was hoping for a release of v2.0.3 before pulling in changes such as this. If the workflows pass, I'll approve.

**Review by wbarnha — APPROVED** [review]
It doesn't fix the 3.12 tests entirely, but nevertheless a step in the right direction. The failing `async` test is a separate issue. Approved!

**Comment by AntonOfTheWoods** [comment]
@wbarnha , can this not get merged without the tests passing, even though the error is unrelated?

**Comment by StephenR-Softcom** [comment]
Thanks for the Pull Request, I am also waiting for the solution so kafka-python works with Python 3.12.0

**Comment by wbarnha** [comment]
Sure, I'll make an exception to merge this.

**Comment by fcs-ts** [comment]
Any chance of getting a point release for this change soon?

**Comment by wbarnha** [comment]
I'll reach out to Dana to see if I can get the ball rolling!

**Comment by Romain-Geissler-1A** [comment]
Hi,

Sorry for jumping on this closed thread, I found it after hitting a similar import issue with six after upgrading to python 3.12 ;) Are there still currently some blocking points preventing to use this package on python 3.10 to 3.12 ? I see that the six package was just one issue, but they might be others (at least for python 3.12).

**Comment by wbarnha** [comment]
> Hi,
> 
> Sorry for jumping on this closed thread, I found it after hitting a similar import issue with six after upgrading to python 3.12 ;) Are there still currently some blocking points preventing to use this package on python 3.10 to 3.12 ? I see that the six package was just one issue, but they might be others (at least for python 3.12).

Not that I know of. Since all tests are now passing for 3.12, I don't see any immediate reason why this wouldn't work. But I should note I personally have not used this on 3.12.

**Comment by StephenR-Softcom** [comment]
> Hi,
> 
> Sorry for jumping on this closed thread, I found it after hitting a similar import issue with six after upgrading to python 3.12 ;) Are there still currently some blocking points preventing to use this package on python 3.10 to 3.12 ? I see that the six package was just one issue, but they might be others (at least for python 3.12).

I have successfully used the library with Python 3.11 on Windows in the last release available from pip.

**Comment by Romain-Geissler-1A** [comment]
Ok, so I will wait for a release to be cut and published through PyPi then. Thanks for your work ;)

**Comment by wbarnha** [comment]
I apologize for taking so with long with this release. I'm still waiting to hear back from the maintainer. You can directly install from this repository via `pip install git+https://github.com/dpkp/kafka-python.git`, however I do not recommend making this a normal trend.

**Comment by wbarnha** [comment]
Hi there, I'm waiting for PyPi credentials. Until then, I am powerless. We have been trying to get in contact with the author, but he has been unresponsive again. ☹️

**Comment by aisven** [comment]
@wbarnha Thank you for your reply. Glad to hear you keep trying after I also read https://github.com/aio-libs/aiokafka/issues/665 a few minutes ago.

**Comment by wbarnha** [comment]
Still trying. Thanks for your patience. :smiling_face_with_tear:

**Comment by skywrathmage006** [comment]
Hi there,
May I know when the release for kafka-python be available for Python 3.11 / 3.12?
Or are there any alternative packages that are equivalent for Python 3.11 / 3.12?

**Comment by wbarnha** [comment]
> Hi there, May I know when the release for kafka-python be available for Python 3.11 / 3.12? Or are there any alternative packages that are equivalent for Python 3.11 / 3.12?

See https://github.com/wbarnha/kafka-python-ng

**Comment by jaredmauch** [comment]
I would also like to 👍 releasing this as I just saw this today in a clean install of ubuntu noble and would prefer to not switch to the other package if this one can be fixed

**Pull request merged**
